airflow/ui/src/pages/Task/Overview/TaskInstanceDuration.tsx

```diff
@@ -0,0 +1,147 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Box, Heading } from "@chakra-ui/react";
+import {
+  Chart as ChartJS,
+  CategoryScale,
+  LinearScale,
+  PointElement,
+  LineElement,
+  BarElement,
+  Filler,
+  Tooltip,
+} from "chart.js";
+import type { PartialEventContext } from "chartjs-plugin-annotation";
+import annotationPlugin from "chartjs-plugin-annotation";
+import dayjs from "dayjs";
+import { Bar } from "react-chartjs-2";
+
+import type { TaskInstanceResponse } from "openapi/requests/types.gen";
+import { system } from "src/theme";
+import { getDuration } from "src/utils/datetime_utils";
+
+ChartJS.register(
+  CategoryScale,
+  LinearScale,
+  PointElement,
+  BarElement,
+  LineElement,
+  Filler,
+  Tooltip,
+  annotationPlugin,
+);
+
+const average = (ctx: PartialEventContext, index: number) => {
+  const values: Array<number> | undefined = ctx.chart.data.datasets[index]?.data as Array<number> | undefined;
+
+  return values === undefined ? 0 : values.reduce((initial, next) => initial + next, 0) / values.length;
+};
+
+export const TaskInstanceDuration = ({
+  taskInstances,
+  totalEntries,
+}: {
+  readonly taskInstances: Array<TaskInstanceResponse> | undefined;
+  readonly totalEntries: number;
+}) => {
+  if (!taskInstances) {
+    return undefined;
+  }
+
+  const runAnnotation = {
+    borderColor: "green",
+    borderWidth: 1,
+    label: {
+      content: (ctx: PartialEventContext) => average(ctx, 1).toFixed(2),
+      display: true,
+      position: "end",
+    },
+    scaleID: "y",
+    value: (ctx: PartialEventContext) => average(ctx, 1),
+  };
+
+  const queuedAnnotation = {
+    borderColor: "grey",
+    borderWidth: 1,
+    label: {
+      content: (ctx: PartialEventContext) => average(ctx, 0).toFixed(2),
+      display: true,
+      position: "end",
+    },
+    scaleID: "y",
+    value: (ctx: PartialEventContext) => average(ctx, 0),
+  };
+
+  return (
+    <Box>
+      <Heading pb={2} size="sm" textAlign="center">
+        Last {totalEntries} task instances
```

Comment:
@bbovenzi Since no filter is applied and the API call just sorts by `logical_date` here `totalEntries` will be the total count of task instances and not 14 though 14 objects are returned. I guess this has to be hardcoded or passed as a constant value to the component in sync with the limit value passed to the API.

Reply:
Thanks, updated.